modules/platform/nuxeo-platform-document-routing/nuxeo-routing-core/src/test/java/org/nuxeo/ecm/platform/routing/test/TestRoutingListener.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * (C) Copyright 2021 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Charles Boidot <cboidot@nuxeo.com>
+ */
+package org.nuxeo.ecm.platform.routing.test;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.Serializable;
+import java.util.Map;
+
+import org.nuxeo.ecm.automation.core.Constants;
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.EventListener;
+import org.nuxeo.ecm.platform.task.TaskEventNames;
+
+public class TestRoutingListener implements EventListener {
+
+    @Override
+    public void handleEvent(Event event) {
+        if (TaskEventNames.WORKFLOW_TASK_REASSIGNED.equals(event.getName())
+                || TaskEventNames.WORKFLOW_TASK_DELEGATED.equals(event.getName())) {
+            Map<String, Serializable> prop = event.getContext().getProperties();
+            Map<String, Serializable> wfVar = (Map<String, Serializable>) prop.get(Constants.VAR_WORKFLOW);
+            Map<String, Serializable> nodeVar = (Map<String, Serializable>) prop.get(Constants.VAR_WORKFLOW_NODE);
+            assertEquals(GraphRouteTest.DUMMY_WF_VAR, wfVar.get("stringfield"));
+            assertEquals(GraphRouteTest.DUMMY_NODE_VAR, nodeVar.get("stringfield2"));
```

Comment:
Doing simple assertions in a sync listener isn't enough, because they'll be swallowed and just logged but won't make the test fail.

You can check this by making the new `getWorkflowContextualInfo` always return an empty `HashMap`, your test will still pass, although logging:

    [main] ERROR [EventServiceImpl] Exception during testListeners sync listener execution, continuing to run other listeners
    java.lang.NullPointerException: null
	at org.nuxeo.ecm.platform.routing.test.TestRoutingListener.handleEvent(TestRoutingListener.java:41) ~[test-classes/:?]

It's crucial that the new test must fail if the fix is removed.

To avoid this the simplest way is probably to add `event.markBubbleException();` at the start of the event handler. It'll make the event service not swallow any exceptions raised in the sync listeners for this event.
Also add `assertNotNull(wfVar)` etc. before the existing assertions because that's how the test actually fails.